### DISCUSSION TITLE
Track MD5s per achievement

### DISCUSSION THREAD

**Discussion by Jamiras** [RetroAchievements/RAWeb#814] [Ideas]
When an achievement is triggered, include the MD5 in the submission (toolkit change). This could be tracked on the backend and used to generate a list of completely compatible MD5s and a list of mostly-compatible MD5s for a game. It would also provide a way to see which MD5s are least commonly used, allowing for easy identification of candidates for unlinking.

See also #1 

For simplicity, recommend counting unlocked achievements per MD5, rather than tracking individual unlocks. 
* Counter should not be incremented if user has already unlocked the achievement (either in softcore more, or due to reactivating it manually). 
* Counters should be reset if achievement is modified to reset compatibility.

**Comment by meleu** [comment]
Just a heads-up:
In [this RAIntegration's PR](https://github.com/RetroAchievements/RAIntegration/pull/242) @Jamiras makes the game hash be included in the unlock request (as `m` parameter). Currently the server simply ignores it, but later it can be used for the feature requested here.

It would be cool to have the same behavior in RetroArch.
